app/src/main/java/com/stevesoltys/seedvault/transport/backup/PackageService.kt

```diff
@@ -29,29 +32,39 @@ private const val LOG_MAX_PACKAGES = 100
 internal class PackageService(
     private val context: Context,
     private val backupManager: IBackupManager,
-) {
+) : KoinComponent {
 
     private val packageManager: PackageManager = context.packageManager
     private val myUserId = UserHandle.myUserId()
+    private val backupCoordinator by inject<BackupCoordinator>()
 
     val eligiblePackages: Array<String>
         @WorkerThread
         @Throws(RemoteException::class)
         get() {
             val packages = packageManager.getInstalledPackages(0)
-                .map { packageInfo -> packageInfo.packageName }
-                .sorted()
+                .sortedBy { it.packageName }
 
             // log packages
             if (Log.isLoggable(TAG, INFO)) {
                 Log.i(TAG, "Got ${packages.size} packages:")
-                packages.chunked(LOG_MAX_PACKAGES).forEach {
-                    Log.i(TAG, it.toString())
-                }
+                logPackages(packages.map { it.packageName })
             }
 
-            val eligibleApps =
-                backupManager.filterAppsEligibleForBackupForUser(myUserId, packages.toTypedArray())
+            // As a result of switching to D2D, we can no longer use backupManager's method
+            // filterAppsEligibleForBackupForUser, because it will not return all of the expected
+            // apps; it is not designed to determine MIGRATION eligibility, only eligibility for
+            // inclusion in *scheduled* BACKUPs, which are implicitly *not* D2D migrations.
+            // None of the other eligibility methods are exposed by AOSP APIs. On the other hand,
+            // the actual backup process properly utilizes OperationType.MIGRATION and performs its
+            // own checks as to whether apps are allowed to be backed up. All we need to do now is
+            // filter out apps that *we* want to be excluded. The system will do the rest later.
+            val eligibleApps = packages.filter {
+                val isFullBackup = if (it.applicationInfo.backupAgentName != null)
+                    it.applicationInfo.flags and FLAG_FULL_BACKUP_ONLY != 0
+                    else true
```

Comment:
Could you reformat this to be a bit more readable? Actually scratch that. We don't even care about is `isFullBackup`. So if we move the `isAppEligibleForBackup()` into another place, we could just remove the `isFullBackup` parameter.

Reply:
Done; it's all moved to `SettingsManager` now.